diracx-client/src/diracx/client/_patch.py

```diff
@@ -113,9 +114,12 @@ def on_request(
 
         if not self._token:
             credentials = json.loads(self._credential.location.read_text())
-            self._token = self._credential.get_token(
-                "", refresh_token=credentials["refresh_token"]
-            )
+            refresh_token = credentials["refresh_token"]
+            if not is_refresh_token_valid(refresh_token):
```

Comment:
would it make sense to print a clear error message, to invite people to login again ?

Reply:
There is the possibility to add the following method in `DiracBearerTokenCredentialPolicy`:

```python
async def on_response(self, request: PipelineRequest, response: PipelineResponse) -> Awaitable[None] | None:
        if response.http_response.status_code == 401:
            print(f"Not authenticated, please log in again: {response.http_response.text()}")
            return

        return super().on_response(request, response)
```

It is executed after each request performed by `DiracClient`. However, I wonder if it makes sense to add the `print` statement in `diracx-client` as it can theoretically be used in various contexts (`diracx-cli`, `diracx-api`).

May be adding a decorator to handle authN/authZ exceptions (401, 403) and print a clear message for the end users of `dirac-cli` could be considered?

Reply:
I think this could be included here:

https://github.com/DIRACGrid/diracx/blob/608487d09dd74dd0003e6518019d6a4dc4303e39/diracx-cli/src/diracx/cli/utils.py#L11

Reply:
https://github.com/DIRACGrid/diracx/pull/241/files#diff-f82a792c5910fa04a412a937ead18e2eb741c28fde1e6c7485b3df198b970ad7R18-R24